src/sentry/event_manager.py

```diff
@@ -1928,6 +1938,106 @@ def _detect_performance_problems(jobs, projects):
         job["performance_problems"] = detect_performance_problems(job["data"])
 
 
+class Performance_Job(TypedDict, total=False):
+    performance_problems: Sequence[PerformanceProblem]
+
+
+@metrics.wraps("save_event.save_aggregate_performance")
+def _save_aggregate_performance(jobs: Sequence[Performance_Job], projects):
+
+    # TODO: batch operations (like rate limiting) so we don't repeat for each job
+    MAX_GROUPS = (
+        10  # safety check in case we are passed too many. constant will live somewhere else tbd
+    )
+    for idx, job in enumerate(jobs):
+        job["groups"] = []
+
+        # General system-wide option
+        rate = options.get("performance.issues.all.problem-creation")
+
+        # More granular, per-project option
+        per_project_rate = projects[idx].get_option("sentry:performance_issue_creation_rate")
```

Comment:
this project option was added here https://github.com/getsentry/sentry/pull/38470